spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/firestore/GcpFirestoreAutoConfiguration.java

```diff
@@ -92,13 +95,19 @@ public class GcpFirestoreAutoConfiguration {
   @Bean
   @ConditionalOnMissingBean
   public FirestoreOptions firestoreOptions() {
-    return FirestoreOptions.getDefaultInstance().toBuilder()
+
+    FirestoreOptions.Builder builder = FirestoreOptions.getDefaultInstance().toBuilder()
         .setCredentialsProvider(this.credentialsProvider)
         .setProjectId(this.projectId)
         .setHeaderProvider(USER_AGENT_HEADER_PROVIDER)
         .setChannelProvider(
-            InstantiatingGrpcChannelProvider.newBuilder().setEndpoint(this.hostPort).build())
-        .build();
+            InstantiatingGrpcChannelProvider.newBuilder().setEndpoint(this.hostPort).build());
+
+    if (databaseId != null) {
```

Comment:
Do we want to check for empty String as well?

Reply:
I don't think it's needed. We check for null across the board in this project in scenarios like this.